packages/@lwc/template-compiler/src/__tests__/fixtures/comments/directive-if/expected.js

```diff
@@ -2,12 +2,20 @@ import { parseFragment, registerTemplate } from "lwc";
 const $fragment1 = parseFragment`<p${3}>true branch</p>`;
 const $fragment2 = parseFragment`<p${3}>false branch</p>`;
 function tmpl($api, $cmp, $slotset, $ctx) {
-  const { co: api_comment, st: api_static_fragment } = $api;
+  const { co: api_comment, st: api_static_fragment, fr: api_fragment } = $api;
   return [
-    $cmp.truthyValue ? api_comment(" HTML comment inside if:true ") : null,
-    $cmp.truthyValue ? api_static_fragment($fragment1(), 1) : null,
-    !$cmp.truthyValue ? api_comment(" HTML comment inside if:false ") : null,
-    !$cmp.truthyValue ? api_static_fragment($fragment2(), 3) : null,
+    $cmp.truthyValue
+      ? api_fragment(2, [
+          api_comment(" HTML comment inside if:true "),
+          api_static_fragment($fragment1(), 1),
+        ])
+      : null,
```

Comment:
This is how I was planning to return the children for the `else-if` directives but with `api_flatten` instead.  This will definitely come in handy for `else-if` :)

Reply:
^ This should be fairly straightforward to resolve for the `else-if` directives then since we're generally matching the overall structure.